library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -179,9 +199,16 @@ - (instancetype)initWithHandle:(envoy_stream_t)handle
     return nil;
   }
 
-  _streamHandle = handle;
+  // Release the stream when the connection closes.
```

Comment:
nit: Release the strong reference to the stream when the ~connection~ stream is closed